ADR/pipeline-service/adr-0001.md

```diff
@@ -0,0 +1,46 @@
+# Service Account configuration for image builds in AppStudio 
+
+Date: Oct 5, 2022
+
+## Status
+
+Proposed
+
+## Context
+
+The `serviceAccount` used for running a `PipelineRun` workload in a workload cluster needs to have the right SCCs and linked credentials to be able to successfully execute a container image build. The linked credentials will mostly originate from the KCP workspace where the `PipelineRun` was synced from.
+Additionally, the SCC association needed for elevating capabilities required for image builds isn't something we'd prefer to expose the user to.
+
+
+
+## Decision
+
+* As a Pipeline Service platform feature, any service/user intending to build a container image would need to create/use the "image-builder" service account in the KCP workspace. 
+Use of any other service accounts will not lead to a successful assembling of the image.
+
+* The AppStudio Build controller would tap into this platform contract to create relevant PipelineRuns for building container images.
+
+## Consequences
+
+
+### Implementation - AppStudio Build Service Controller
+
+* The AppStudio Build controller creates an "image-builder" service account in the KCP workspace/namespace.
```

Comment:
By default `builder` SA is created on compute clusters, should not we use name connected with pipeline-service? eg. `pipeline-service-builder` ?

Reply:
I agree this should is something that should be tied to the Pipeline Service. Part of what the Pipeline Service should provide is a service account capable of executing builds with reasonably elevated pod security permissions.

Something to consider - would a future user or on-top service want this disabled? For example - a service wants to use Pipeline Service as a workflow orchestrator, and does not want the ability to build container images. We've seen this situation play out with OpenShift Builds, where customers want to disable things that could be a security risk.

I think this is feasible with an API that the pipeline service workspace controller reconciles.

Reply:
> Something to consider - would a future user or on-top service want this disabled? For example - a service wants to use Pipeline Service as a workflow orchestrator, and does not want the ability to build container images. We've seen this situation play out with OpenShift Builds, where customers want to disable things that could be a security risk.

I'd say, whoever owns the workload cluster has a greater reason to worry and therefore is the one taking the call on who gets to use the feature.